firmware/repository.py

```diff
@@ -0,0 +1,126 @@
+"""Repository - module to communicate with the database"""
+
+from firmware.database import db_session
+from firmware.models import User, Company, Category, Review
+from sqlalchemy import and_
+
+
+def get_companies():
+    """ gets all companies from database """
+    companies = db_session.query(Company)
+    return companies
+
+
+def get_user_id(username):
+    """ gets a user id via it's unique username """
+    user_id = db_session.query(User.id).filter(User.username == username)
+    return user_id
+
+
+def add_company(company):
+    """ adds a company to the database """
+    db_session.add(company)
+    db_session.commit()
+    return company.id
+
+
+def get_company(company_id):
+    """ retrieves a specific company, via it's id """
+    company = db_session.query(Company).filter(Company.id == company_id).one()
+    if company is None:
+        return None
+    return company
+
+
+def get_category(company_id):
```

Comment:
The company should have a field called `category` and you should access the category using that, i.e. `company.category`.